capture/scopes/chipwhisperer/husky.py

```diff
@@ -0,0 +1,106 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import time
+
+import chipwhisperer as cw
+import numpy as np
+from scopes.chipwhisperer.cw_segmented import CwSegmented
+
+SAMPLING_RATE_MAX = 200e6
+
+
+class Husky:
+    def __init__(self, scope_gain, num_cycles, num_segments, offset_cycles,
+                 pll_frequency, target_clk_mult):
+        self.scope = None
+        self.scope_gain = scope_gain
+        # In our setup, Husky operates on the PLL frequency of the target and
+        # multiplies that by an integer number to obtain the sampling rate.
+        # Note that the sampling rate must be at most 200 MHz.
+        self.clkgen_freq = pll_frequency
+        self.adc_mul = int(SAMPLING_RATE_MAX // pll_frequency)
+        self.sampling_rate = self.clkgen_freq * self.adc_mul
```

Comment:
:) i think the computed sampling rate will always equal SAMPLING_RATE_MAX, right?

Reply:
in other words: sampling rate here is fixed at max but seems to be computed from inputs. There is no cfg param atm to adjust sampling rate. it will always be max.

Reply:
We could have the sampling rate as a cfg param and compute adc_mul from it similar to above.

Reply:
I agree, from the usability side, it would make sense for me that the user needs to specify `pll_frequency`, `target_freq` (user should now at which freq. OT is running), and desired `sampling_rate`. So the code would look like this:

```python
        self.scope = None
        self.scope_gain = scope_gain
        # In our setup, Husky operates on the PLL frequency of the target and
        # multiplies that by an integer number to obtain the sampling rate.
        # Note that the sampling rate must be at most 200 MHz.
        self.clkgen_freq = pll_frequency
        self.sampling_rate =  sampling_rate
        self.adc_mul = self.sampling_rate / self.clkgen_freq

        # The target runs on the PLL clock but uses internal clock dividers and
        # multiplier to produce the clock of the target block.
        self.target_freq = target_freq

        # The scope is configured in terms of samples. For Husky, the number of
        # samples must be divisble by 3 for batch captures.
        sampling_target_ratio = self.sampling_rate / self.target_freq
        self.offset_samples = int(offset_cycles * sampling_target_ratio)
        self.num_samples = int(num_cycles * sampling_target_ratio)
        if self.num_samples % 3:
            self.num_samples = self.num_samples + 3 - (self.num_samples % 3)

        self.num_segments = num_segments
        self.scope = None
```

Reply:
i think this would be great, and prevent that we would need to hand pll_freq to a waverunner scope as well

Reply:
thx - understand that there are limitations here. Maybe making those limitations explicit as comments along with params which have self-explanatory names is still easier? :)

Reply:
We could sanity check the user input to avoid that `adc_mul `isn't an integer.

Reply:
We quickly discussed offline that in the future the user should define `target_clk_freq` and `sampling_rate` and that helper function should then do the magic and throw an error / adjust + warn if the resulting `adc_mul` isn't integer. We've also said that this can be done in a follow-up PR of this one.